src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -227,6 +229,77 @@ module.exports = class LevelPlane extends Array {
       this[myIndex].blockType = "redstoneWireCross";
     }
 
+    if (this[myIndex].isPowered) {
+      this[myIndex].blockType += "On";
+    }
+
     return this[myIndex].blockType;
   }
+
+  getRedstone() {
+    this.redstoneList = [];
+    this.redstoneListON = [];
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType.substring(0, 12) === "redstoneWire") {
+        this[i].isPowered = false;
+        let position = this.indexToCoordinates(i);
+        this.redstoneList.push(position);
+      }
+    }
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType === "railsRedstoneTorch") {
+        let position = this.indexToCoordinates(i);
+        this.redstonePropagation(position);
+      }
+    }
+
+    for (let i = 0; i < this.redstoneList.length; ++i) {
+      this.determineRedstoneSprite(this.redstoneList[i]);
+    }
+    if (this.levelModel !== null) {
+      this.levelModel.controller.levelView.refreshActionPlane(this.redstoneList);
```

Comment:
Does this mean we potentially refresh the same location twice?  (Once here, and once on line 76.)

As I've said before, I think this has the potential to introduce a lot of hard-to-track down bugs.  I’d prefer for the LevelPlane to have free reign changing blocks however it sees fit, then there’s one method at the end that compares the current state to the state of the sprites and does a batch update of the LevelView.

Reply:
Ah, yes. That's true. 76 should only be refreshing stuff if it -wasn't- a redstone block.

I agree that something more systemic in terms of comparing all blockTypes and refreshing only those indices at the end that don't match "last tick" or what not would be a fine idea. But I didn't know if that seemed like a change a bit more independent of the redstone functionality itself (it would certainly effect redstone, but it would effect every other placement/destruction/change to a plane in general so it didn't feel like this was the place?)

Reply:
If we're not doing that cleanup now, at minimum can we have this function return the list of positions that should be refreshed?

This way we're still calling the view refresh in a single centralized place.

Reply:
That should be do-able. Let me play with some places that make sense in a way that won't disrupt behavior.